File: src/main/java/ch/grassl/common/AdventOfCodeDay.java
package ch.grassl.common;

import lombok.Getter;
import lombok.extern.slf4j.Slf4j;

import java.util.LinkedList;

@Slf4j
public abstract class AdventOfCodeDay {

    @Getter
    protected final LinkedList<Object> result = new LinkedList<>();

    public abstract void run();

    public void printResult() {
        String day = this.getClass().getSimpleName();
        log.info("----------- {} ------------", day);
        for (Object r : result) {
            log.info("{}: The {}. result is: {}", day, result.indexOf(r) + 1, r);
        }
    }
}

File: src/main/java/ch/grassl/common/AdventOfCodeDay.kt
package ch.grassl.common

import org.slf4j.LoggerFactory
import java.util.*

abstract class AdventOfCodeDay {

    private val log = LoggerFactory.getLogger(this.javaClass)

    val result = LinkedList<Any>()

    abstract fun run()

    fun printResult() {
        val day = this.javaClass.simpleName
        log.info("----------- {} ------------", day)
        result.forEach {
            log.info("{}: The {}. result is: {}", day, result.indexOf(it) + 1, it)
        }
    }
}
